src/v/cloud_storage/tests/topic_mount_handler_test.cc

```diff
@@ -197,8 +203,10 @@ TEST_P(TopicMountHandlerFixture, TestMountTopicSuccess) {
 
     auto handler = topic_mount_handler(bucket_name, remote.local());
 
-    auto mount_result = handler.mount_topic(topic_cfg, rtc).get();
-    ASSERT_EQ(mount_result, topic_mount_result::success);
+    auto prepare_result = handler.prepare_mount_topic(topic_cfg, rtc).get();
+    ASSERT_EQ(prepare_result, topic_mount_result::success);
```

Comment:
Before, when `mount_topic` was a one shot operation, `topic_mount_result::success` would be an indication that the topic was indeed mounted. Now, it can have two meanings: for a `check` op, it means that the topic CAN be mounted, and for a `confirm` op, that the topic was mounted.

I don't think we should define both cases with `topic_mount_result::success`.